src/App.js

```diff
@@ -28,31 +28,94 @@ const generateSquares = () => {
 const App = () => {
 
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
+
+
+  const switchPlayers = () => {
+    if (player === PLAYER_1) {
+      setPlayer(PLAYER_2)
+    }
+    if (player === PLAYER_2) {
+      setPlayer(PLAYER_1)
+    }; 
+  };
+
+ 
+
 
   // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  let onClickCallback = (squareClickedOnId) => {
+    let updatedSquares = []
+    
+    if (checkForWinner() !== null) {
+      return; 
+    }
+    
+    squares.forEach((row) => {
+      let updatedSquareRow = []
+
+      row.forEach((square) => {
+        if (squareClickedOnId === square.id){
+          if (square.value === ''){
+          const updatedSquare = {id: squareClickedOnId, value: player }
+          updatedSquareRow.push(updatedSquare)
+          switchPlayers()
+          } else {
+            updatedSquareRow.push(square) 
+          }
+        }else {
+          updatedSquareRow.push(square)
```

Comment:
This is an appropriate spot for a `;` as the end of the line marks the end of a statement.